### DISCUSSION TITLE
502 Bad Gateway error on login in a second tab

### DISCUSSION THREAD

**Discussion by spiderdab** [keycloak/keycloak#32430] [Q&A]
keycloak version 24.0.5

I searched for this error but only found issues on the header size, but I don't think this is the problem here.

Login works ok in a first tab.
If I try to login in a second tab, in the same browser, I instantly receive a 502 Bad Gateway error (before the KC login screen.)
If I close the user session from the KC admin page, I can login again from my browser..

A second login from a private tab works ok.
My keycloak admin migrated the keycloak installation from version 16.1 and I didn't have this problem before.

What can it be?

**Comment by codespearhead** [comment]
I cannot reproduce it in the latest stable release of Keycloak, which currently is version `26.0.0`.

All similar error reports have been closed because the issue could not be reproduced [1]. It is very likely that the problem is specific to your infrastructure, possibly due to a reverse proxy misconfiguration [2][3][4][5].

[1] https://github.com/search?q=repo%3Akeycloak%2Fkeycloak+%22Bad+Gateway%22&type=issues
[2] https://github.com/keycloak/keycloak/issues/33414
[3] https://github.com/keycloak/keycloak/issues/33414#issuecomment-2389657317
[4] https://github.com/keycloak/keycloak/discussions/22605
[5] https://github.com/keycloak/keycloak/discussions/22605#discussioncomment-10931827

**Reply by indu-devbala** [reply]
@spiderdab do you solve this issue .

i too facing the same issue. if you solved can you please share me the solution

**Reply by codespearhead** [reply]
As of 2025-01-09, with the latest stable Keycloak version being `26.0.7`, my earlier comment [1] remains relevant [2]: you're likely using deprecated env vars (reverse proxy misconfiguration) and that's breaking the authorization flow.

[1] https://github.com/keycloak/keycloak/discussions/32430#discussioncomment-10950599
[2] https://github.com/search?q=repo%3Akeycloak%2Fkeycloak+%22502+Bad+Gateway%22&type=issues

**Reply by spiderdab** [reply]
> @spiderdab do you solve this issue .
> 
> i too facing the same issue. if you solved can you please share me the solution

In my case (IIRC) I solved saving the access-token in the localStorage instead of the sessionStorage.
